modules/effects/src/concat_latest_from.ts

```diff
@@ -38,24 +38,19 @@ export function concatLatestFrom<T extends Observable<unknown>, V>(
  * ```
  */
 export function concatLatestFrom<
-  T extends Observable<unknown>[] | Observable<unknown>,
+  T extends ObservableInput<unknown>[] | ObservableInput<unknown>,
   V,
   R = [
     V,
-    ...(T extends Observable<unknown>[]
+    ...(T extends ObservableInput<unknown>[]
       ? { [i in keyof T]: ObservedValueOf<T[i]> }
       : [ObservedValueOf<T>])
   ]
 >(observablesFactory: (value: V) => T): OperatorFunction<V, R> {
-  return pipe(
-    concatMap((value) => {
-      const observables = observablesFactory(value);
-      const observablesAsArray = Array.isArray(observables)
-        ? observables
-        : [observables];
-      return of(value).pipe(
-        withLatestFrom(...observablesAsArray)
-      ) as unknown as Observable<R>;
-    })
+  return concatMap(
+    (value) =>
+      forkJoin(observablesFactory(value)).pipe(
```

Comment:
I'm not sure if `forkJoin` is the right fit here because it waits for all observables to complete.

The `concatLatestFrom` operator is usually used with state observables that never complete:

```ts
concatLatestFrom(() =>
  this.store.select(selectFoo) // this observable never completes
)
```